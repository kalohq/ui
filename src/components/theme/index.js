/* @flow */
export {default} from './theme';
