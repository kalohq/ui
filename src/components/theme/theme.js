/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
