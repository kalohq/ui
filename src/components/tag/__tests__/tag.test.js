/* @flow */
/* eslint-env jest */
import React from 'react';
import {sheet} from 'emotion';
import {mount} from 'enzyme';
import serializer from 'jest-glamor-react';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import theme from '../../theme';

import Tag from '../tag';

expect.addSnapshotSerializer(serializer(sheet));

describe('components/tag', () => {
  const create = (props = {}) =>
    renderer
      .create(
        <ThemeProvider theme={theme}>
          <Tag {...props} />
        </ThemeProvider>
      )
      .toJSON();

  it('should shallow render as expected', () => {
    const element = create();
    expect(element).toMatchSnapshot();
  });

  it('should render with a remove icon if an onRemove function is passed', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Tag theme="orange" label="Front-end Engineer" onRemove={() => {}} />
      </ThemeProvider>
    );

    expect(wrapper.find('Icon').length).toBe(1);
  });
});
