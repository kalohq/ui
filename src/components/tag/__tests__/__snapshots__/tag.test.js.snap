// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/tag should shallow render as expected 1`] = `
.glamor-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 28px;
  -webkit-transition: color 0.5s ease-in;
  transition: color 0.5s ease-in;
  background: rgb(193,244,255);
  border: 1px solid currentColor;
  color: rgb(77,184,207);
  border-radius: 16px;
  padding: 0 16px;
  vertical-align: middle;
  max-width: 100%;
  width: fit-content;
  cursor: default;
}

.glamor-1 {
  display: inline-block;
  vertical-align: bottom;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 12px;
  color: currentColor;
}

.glamor-1:hover {
  text-decoration: inherit;
}

.glamor-1 strong {
  font-weight: 500;
}

<span
  className="glamor-2 glamor-3"
  onClick={undefined}
  style={Object {}}
>
  .glamor-1 {
  display: inline-block;
  vertical-align: bottom;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 12px;
  color: currentColor;
}

.glamor-1:hover {
  text-decoration: inherit;
}

.glamor-1 strong {
  font-weight: 500;
}

<span
    className="glamor-0 glamor-1"
    color="currentColor"
    onClick={undefined}
    size="extra-small"
    style={Object {}}
  />
</span>
`;
